railib/rest.py

```diff
@@ -210,6 +214,28 @@ def _authenticate(ctx: Context, req: Request) -> Request:
         return req
     raise Exception("unknown credential type")
 
+# Issues an HTTP request and retries if failed due to URLError.
+def _urlopen_with_retry(req: Request, retries: int = 0):
+    if retries < 0:
+        raise ValueError("Retries must be a non-negative integer")
```

Comment:
I would add this to the user entry-point (i.e. Context construction) for this to be more comprehensible